extensions/Worker.Extensions.ServiceBus/src/ServiceBusSessionMessageActionsConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using Microsoft.Azure.ServiceBus.Grpc;
+using System.Text.Json;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Converter to bind to <see cref="ServiceBusSessionMessageActions" /> or <see cref="ServiceBusSessionMessageActions[]" /> type parameters.
+    /// </summary>
+    [SupportsDeferredBinding]
+    [SupportedTargetType(typeof(ServiceBusSessionMessageActions))]
+    [SupportedTargetType(typeof(ServiceBusSessionMessageActions[]))]
+    internal class ServiceBusSessionMessageActionsConverter : IInputConverter
+    {
+        private readonly Settlement.SettlementClient _settlement;
+
+        public ServiceBusSessionMessageActionsConverter(Settlement.SettlementClient settlement)
+        {
+            _settlement = settlement;
+        }
+
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            try
+            {
+                var foundSessionId = context.FunctionContext.BindingContext.BindingData.TryGetValue("SessionId", out object? sessionId);
+                if (!foundSessionId)
+                {
+                    throw new InvalidOperationException("Expecting SessionId within binding data and value was not present.");
+                }
+
+                // Get the sessionLockedUntil property from the SessionActions binding data
+                var foundSessionActions = context.FunctionContext.BindingContext.BindingData.TryGetValue("SessionActions", out object? sessionActions);
+                if (!foundSessionActions)
+                {
+                    throw new InvalidOperationException("Expecting SessionActions within binding data and value was not present.");
```

Comment:
For the other error conditions here, is there any additional information we can provide? Fine if not (as this would still be used for troubleshooting), but if we have the ability to share more information about the causes, let's add that.

Reply:
I'm not sure if we can add any more info to these error messages. I think the primary root cause for these errors would be is isSessionEnabled is not set for the queue, but the user would hit the error in one of the earlier error messages before.